pkg/ipcache/metadata.go

```diff
@@ -639,48 +641,13 @@ func (ipc *IPCache) doInjectLabels(ctx context.Context, modifiedPrefixes []cmtyp
 		}
 	}
 	if len(idsToDelete) > 0 {
-		if ipc.UpdatePolicyMaps(ctx, nil, idsToDelete) {
-			mutated = true
-		}
+		ipc.IdentityUpdater.UpdateIdentities(nil, idsToDelete)
```

Comment:
Why are we no longer waiting for deleted identities to propagate? Is this race-free?

Reply:
Very good catch! I don't know what I was thinking. We absolutely have to wait here.